perses/app/fah_generator.py

```diff
@@ -0,0 +1,203 @@
+    neq_setup_options = yaml.load(neq_yaml_file, Loader=yaml.FullLoader)
+    neq_yaml_file.close()
+
+    _logger.info(f"Running setup...")
+    setup_dict = run_setup(setup_options, serialize_systems=False, build_samplers=False)
+
+    trajectory_directory = setup_options['trajectory_directory']
+
+    n_equilibration_iterations = setup_options['n_equilibration_iterations'] #set this to 1 for neq_fep
+    _logger.info(f"Equilibration iterations: {n_equilibration_iterations}.")
+
+    topology_proposals = setup_dict['topology_proposals']
+    htfs = setup_dict['hybrid_topology_factories']
+
+    #create solvent and complex directories
+    for phase in htfs.keys():
+        dir = os.path.join(os.getcwd(), f"{trajectory_directory}_{phase}", f"RUN_{format(run_number, '03')}")
+        os.mkdir(dir)
+
+        #serialize the hybrid_system
+        data.serialize(htfs[phase].hybrid_system, f"{dir}/system.xml")
+
+        #make and serialize an integrator
+        integrator = make_neq_integrator(setup_options, neq_setup_options, alchemical_functions = DEFAULT_ALCHEMICAL_FUNCTIONS)
+        data.serialize(htfs[phase].hybrid_system, f"{dir}/integrator.xml")
+
+        #create and serialize a state
+        try:
+            state = relax_structure(setup_dict['temperature'],
+                            system = htfs[phase].hybrid_system,
+                            positions = htfs[phase].hybrid_positions,
+                            nminimize=100,
+                            nequil = 4,
+                            n_steps_per_iteration=250)
+
+            data.serialize(state, f"{dir}/state.xml")
+        except Exception as e:
+            print(e)
+            passed=False
+        else:
+            passed=True
+
+
+        #lastly, make a core.xml
+        #TODO: make core.xml
+
+        #create a logger for reference
+        references = {'start_ligand': setup_dict['old_ligand_index'],
+                      'end_ligand': setup_dict['new_ligand_index'],
+                      'protein_filename': setup_dict['protein_pdb'],
+                      'passed_strucutre_relax': passed}
+        with open(f"{dir}/reference.pkl", 'wb') as f:
+            pickle.dump(references, f)
+
+
+if __name__ == "__main__":
+    setup_yaml, neq_setup_yaml, run_number = sys.argv[1], sys.argv[2], sys.argv[3] #define args
+
+    #open the setup yaml to pull trajectory_directory
+    yaml_file = open(setup_yaml, 'r')
+    setup_options = yaml.load(yaml_file, Loader=yaml.FullLoader)
+    yaml_file.close()
+    traj_dir = setup_options['trajectory_directory']
+
+    gather all
+    phases = setup_options['phases']
+    for phase in phases:
+        now_path = os.path.join(os.getcwd(), f"{traj_dir}_{phase}")
+        if not os.path.exists(now_path):
+            os.mkdir(now_path)
+        else:
+            raise Exception(f"{now_path} already exists.  Aborting.")
+    run_neq_fah_setup(setup_yaml, neq_yaml, int(run_number), **kwargs)
```

Comment:
Where do the `kwargs` come from? Should we pull them from the YAML so that we can also configure the switching time, etc?